builtin/credential/aws/path_login.go

```diff
@@ -208,13 +206,22 @@ func (b *backend) pathLoginUpdate(
 		return nil, err
 	}
 
+	clientNonce := data.Get("nonce").(string)
+	if clientNonce == "" && !storedIdentity.DisallowReauthentication {
```

Comment:
Also I think you need to be checking the image and tag disallow flags here. So maybe can the nonce be checked after? Or the tag parsed earlier?